### PR TITLE
chore: Clarify the meaning of secret exfiltration via ApplicationSet (issue #18560)

### LINKED ISSUE #18560: clarify scp-provider appset-in-any-namespace secrets considerations
# Summary

 I read through https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Appset-Any-Namespace/#scm-providers-secrets-consideration and because of this warning about the secret for a SCM Provider I though this provider searches for this secret in the argocd namespace. However, I noticed today that this secret must be placed in the same namespace as the appset, so I ask myself if this warning "By allowing ApplicationSet in any namespace you must be aware that any secrets can be exfiltrated using scmProvider or pullRequest generators." is really accurate since you can only get secrets from your "own" namespace.
Or do I miss something here?

If not, I propose that in the documentation this gets clarified, that only secrets in this special namespace can get exposed, which is actually a namespace exactly for this team?

# Motivation

Then the risks for appset-in-any-namespace with SCMs gets clearer

# Proposal

Adding infos that since secrets for the SCM provider need to get placed in the dev-teams namespace where the appset gets created and the secret probably also gets created by this dev-team, the problem is not that high.

### PR DESCRIPTION
**Pull request by thecooldrop**
Closes #18560 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/operator-manual/applicationset/Appset-Any-Namespace.md [hunk 1]
```diff
@@ -53,7 +56,7 @@ spec:
   template:
 ```
 
-Therefore administrator must restrict the urls of the allowed SCM Providers (example: `https://git.mydomain.com/,https://gitlab.mydomain.com/`) by setting the environment variable `ARGOCD_APPLICATIONSET_CONTROLLER_ALLOWED_SCM_PROVIDERS` to argocd-cmd-params-cm `applicationsetcontroller.allowed.scm.providers`. If another url is used, it will be rejected by the applicationset controller.
+In order to prevent the scenario about administrator must restrict the urls of the allowed SCM Providers (example: `https://git.mydomain.com/,https://gitlab.mydomain.com/`) by setting the environment variable `ARGOCD_APPLICATIONSET_CONTROLLER_ALLOWED_SCM_PROVIDERS` to argocd-cmd-params-cm `applicationsetcontroller.allowed.scm.providers`. If another url is used, it will be rejected by the applicationset controller.
```
Comment (ishitasequeira) [inline docs/operator-manual/applicationset/Appset-Any-Namespace.md:59]: nit: replace `about`  to `above,`

**Review by ishitasequeira — COMMENTED** [review]
Overall the PR LGTM. Just one nit

File: docs/operator-manual/applicationset/Appset-Any-Namespace.md (hunk 1 above)
Comment (thecooldrop) [reply docs/operator-manual/applicationset/Appset-Any-Namespace.md]: Fair nit. Fixed and pushed

**Review by ishitasequeira — COMMENTED** [review]
@thecooldrop , thanks for the changes. Can you fix the DCO?

**Comment by thecooldrop** [comment]
> @thecooldrop , thanks for the changes. Can you fix the DCO?

Hey, sorry I am on somewhat longer vacation and a bit of digital detox and have no laptop. Otherwise I hereby declare that I volutarily donate all my contributions contained in this PR to ArgoCD project without option backing out of my commitment (or some other legal lingo).

If this is not good enough I will bw changing the signatures in about 10ish days from now

**Comment by thecooldrop** [comment]
@ishitasequeira I have signed-off the commits. This should now be mergeable

**Review by ishitasequeira — APPROVED** [review]

**Pull request merged**